app/models/label/LabelTable.scala

```diff
@@ -390,17 +392,32 @@ object LabelTable {
         |  				LEFT JOIN sidewalk.label_severity as sev ON lb.label_id = sev.label_id
         |				  LEFT JOIN sidewalk.label_description as lab_desc ON lb.label_id = lab_desc.label_id
         |				  LEFT JOIN sidewalk.label_temporariness as lab_temp ON lb.label_id = lab_temp.label_id
-        |		  ) AS lb_big
-        |WHERE lb1.deleted = FALSE
+        |     ) AS lb_big,
+        |     (
+        |        SELECT label.label_id, text, COUNT(label_validation_id) AS count
+        |        FROM label
+        |        FULL JOIN validation_options ON TRUE
+        |        LEFT JOIN label_validation ON label.label_id = label_validation.label_id
+        |            AND label_validation.validation_result = validation_options.validation_option_id
+        |        WHERE label.deleted = FALSE
+        |        GROUP BY label.label_id, validation_option_id
+        | ) AS val
+        | WHERE lb1.deleted = FALSE
         |    AND lb1.gsv_panorama_id = gsv_data.gsv_panorama_id
         |    AND lb1.audit_task_id = at.audit_task_id
         |    AND lb1.label_id = lb_big.label_id
         |    AND at.user_id = u.user_id
         |    AND lb1.label_id = lp.label_id
-        |ORDER BY lb1.label_id DESC
-        |LIMIT ?""".stripMargin
+        |    AND lb1.label_id = val.label_id
+        | ORDER BY lb1.label_id DESC
+        | LIMIT ?""".stripMargin
     )
-    selectQuery(takeN).list.map(label => labelAndTagsToLabelMetadata(label, getTagsFromLabelId(label._1)))
+    val lst = selectQuery(takeN).list;
+    val allMetadata = lst.groupBy(_._1).map{ case (y,group) => (y, group.head._2, group.head._3, group.head._4, group.head._5
+    , group.head._6, group.head._7, group.head._8, group.head._9, group.head._10, group.head._11, group.head._12, group.head._13
+    , group.head._14, group.head._15, group.head._16, group.head._17, group.head._18, group.head._19, group.head._20
+    , group.map(z => (z._21, z._22)).toMap)}.toList;
```

Comment:
To make the formatting a bit nicer and consistent with the rest of the code, I would recommend something like this:
```
    val allMetadata = lst.groupBy(_._1).map { case (y,group) => (
      y, group.head._2, group.head._3, group.head._4, group.head._5, group.head._6, group.head._7, group.head._8,
      group.head._9, group.head._10, group.head._11, group.head._12, group.head._13, group.head._14, group.head._15,
      group.head._16, group.head._17, group.head._18, group.head._19, group.head._20,
      group.map(z => (z._21, z._22)).toMap
      )}.toList
```
A few other style things:
* Make sure to keep lines under 120 characters
* Do you need the .toList? I thought this would already be a list anyway.
* Can you rename the y/z to be more descriptive? Maybe something like "id" "labelId" for "y" and just "label" for z.

Reply:
Much better. Still a couple things to address here:
1. You need to indent the parameter list.
1. Again, do we need the .toList?

Reply:
Yea, so without the `.toList` its an immutable collection, which isnt being recognized as a list.

Reply:
Interesting! Not what I expected! Thanks